framework/src/main/java/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -1116,66 +1123,65 @@ private boolean shouldPerformWholeProgramInference(Tree tree, Element elt) {
-        processPostconditionsAndConditionalPostconditions(
-                invocationNode, invocationTree, store, null, postconditions);
+        processPostconditionsAndConditionalPostconditions(n, tree, store, null, postconditions);
     }
 
     /**
      * Add information from the conditional postconditions of a method to the stores after an
      * invocation.
      *
-     * @param invocationNode a method call
+     * @param n a method call or an object creation node
      * @param methodElement the method being called
-     * @param invocationTree the tree for the method call
+     * @param tree the tree for the method call or for the object creation
      * @param thenStore the "then" store; is side-effected by this method
      * @param elseStore the "else" store; is side-effected by this method
      */
     protected void processConditionalPostconditions(
-            MethodInvocationNode invocationNode,
-            ExecutableElement methodElement,
-            Tree invocationTree,
-            S thenStore,
-            S elseStore) {
+            Node n, ExecutableElement methodElement, Tree tree, S thenStore, S elseStore) {
         ContractsFromMethod contractsUtils = analysis.atypeFactory.getContractsFromMethod();
         Set<ConditionalPostcondition> conditionalPostconditions =
                 contractsUtils.getConditionalPostconditions(methodElement);
         processPostconditionsAndConditionalPostconditions(
-                invocationNode, invocationTree, thenStore, elseStore, conditionalPostconditions);
+                n, tree, thenStore, elseStore, conditionalPostconditions);
     }
 
     /**
      * Add information from the postconditions and conditional postconditions of a method to the
      * stores after an invocation.
      *
-     * @param invocationNode a method call
-     * @param invocationTree the tree for the method call
+     * @param n a method call node or an object creation node
+     * @param tree the tree for the method call or for the object creation
      * @param thenStore the "then" store; is side-effected by this method
      * @param elseStore the "else" store; is side-effected by this method
      * @param postconditions the postconditions
      */
     private void processPostconditionsAndConditionalPostconditions(
-            MethodInvocationNode invocationNode,
-            Tree invocationTree,
-            S thenStore,
-            S elseStore,
-            Set<? extends Contract> postconditions) {
+            Node n, Tree tree, S thenStore, S elseStore, Set<? extends Contract> postconditions) {
+
+        StringToJavaExpression stringToJavaExpr = null;
+        if (n instanceof MethodInvocationNode) {
+            stringToJavaExpr =
+                    stringExpr ->
+                            StringToJavaExpression.atMethodInvocation(
+                                    stringExpr, (MethodInvocationNode) n, analysis.checker);
+        }
```

Comment:
Make this an `else if` and throw an exception if it is neither a method invocation nor an object creation.